tools/src/testcert-gen/main.go

```diff
@@ -0,0 +1,302 @@
+package main
+
+import (
+	"context"
+	crypto_rand "crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"math/big"
+	math_rand "math/rand"
+	"net"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/spf13/pflag"
+
+	"github.com/datawire/dlib/dlog"
+)
+
+func djb2Hash(str string) int64 {
+	hash := uint64(5381)
+	for _, c := range str {
+		hash = (hash * 33) + uint64(c)
+	}
+	return int64(hash)
+}
+
+func NewRand(seed string) io.Reader {
+	return &randReader{
+		inner: math_rand.New(math_rand.NewSource(djb2Hash(seed))),
+	}
+}
+
+type randReader struct {
+	inner io.Reader
+}
+
+func (r *randReader) Read(d []byte) (int, error) {
+	if len(d) == 1 {
+		// We want deterministic keys.  But rsa.GenerateMultiPrimeKey calls
+		// crypto/internal/randutil.MaybeReadByte() to make them non-deterministic even when
+		// the source of randomness is deterministic.  Shut that down!
```

Comment:
This should duplicate, or at minimum refer to, your comment in `Main` that talks about why this isn't just a godawful terrible bad idea. 🙂